src/modding/ModLoader.cs

```diff
@@ -167,17 +366,18 @@ public void LoadMods()
     {
         var newMods = Settings.Instance.EnabledMods.Value.ToHashSet();
 
-        foreach (var unload in loadedMods.Where(m => !newMods.Contains(m)).ToList())
+        foreach (var unload in loadedMods.ToList())
```

Comment:
Why? That does not make sense to me if the player wants to load their mods like this:
`1, 2, 3, 4`
and it's was like this:
`1, 3, 2, 4`

then only unloading the ones in the wrong order would produce a load order like this:
`1, 4, 3, 2`
Which is completely different.

I could change it so that that only mods after the first wrong order are unloaded (so in that example everything after 1 is unloaded), that should improve performance or something...

Reply:
> I could change it so that that only mods after the first wrong order are unloaded (so in that example everything after 1 is unloaded), that should improve performance or something...

I guess that makes sense. I was thinking of the situation where mods that don't say they are incompatible with a specific load order, should hopefully work with a slightly different order. Because unloading asset mods is going to be expensive as a full refresh of the game loaded .pcks is needed. But I guess for absolute clarity to the player, you are right about what needs to be done.